index.js

```diff
@@ -147,44 +199,62 @@ const line = []
 
 // ***** Scope & Closures - Question 1 *****
 
+let ticket = 0
 
-// *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+function takeATicketNumber(line) {
+  ticket++
+  line.push(ticket)
+  return `Welcome. You are ticket number ${ticket}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+// *** Uncomment the lines below to test
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 
+function ticketNumberGeneratorFunc(line) {
+  ticket = 0
+  return function takeATicketNumber(line) {
+    ticket++
+    line.push(ticket)
+    return `Welcome. You are ticket number ${ticket}`
+  }
+}
```

Comment:
Nice work on this! Just be careful on line 232 here, if you declare a variable without `let` or `const` it will become a global variable (even if you declare it inside a function). To make sure that the `ticket` variable here is scoped to the function, use `let`.

Reply:
Woops!